accepted/2022/version-selection.md

```diff
@@ -0,0 +1,449 @@
+# .NET 7 Version Selection Improvements
+
+**Owner** [Rich Lander](https://github.com/richlander)
+
+Each development platform is based on some composition model for platform versions (where they are installed, private vs global installation/visibility, how they are selected, ...). On one hand, there are platforms like Node.js that have a single version in scope at a time (tools like [`nvm`](https://github.com/nvm-sh/nvm) make that more manageable) and on the other we have .NET that allows for multiple versions in scope at a time (managed by the [`dotnet`](https://docs.microsoft.com/dotnet/core/tools/dotnet) tool and MSBuild). .NET offers a multiple ways of selecting a runtime or SDK version, however, there are some inconsistencies across the various gestures and missing capabilities. This document proposes improvements to .NET version selection.
+
+Summary of significant changes:
+
+- Disable multi-level lookup.
+- Change default for `global.json` SDK version to a floor version, as opposed to the current pinning behavior.
+- Enable controlling SDK version by environment variable (ENV).
+- Enable proxying to a local/private SDK via the `PATH` SDK.
+- Enable specifying a floor runtime version for an environment.
+
+Note: The last two of those changes are more speculative (less likely to happen).
+
+Let's start by considering some concrete customer scenarios to improve (some of which are overlapping).
+
+## Dev inner loop consistency with lab builds
+
+For many teams, infra builds are the center of their environment. Infra is intended to mean any of Continuous Integration (CI), Pull Request (PR), or Official builds. Teams typically put the most design and implementation effort into their infra builds, with the goal of establishing their most correct and reliable environment. This includes defining their SDK and runtime platform dependencies. In other words, if there is a bug that repros in CI but not on your local dev box, you still have a bug, while the opposite might not be true.
+
+Most devs would prefer a local environment that matches their lab environment. The practice of working in matching environments (local vs lab and across teammates) increases productivity since it inherently avoids spurious local issues or surprise problems in PR builds. There is no good guidance or productized UX that would help teams establish matching environments.
```

Comment:
I'd argue that most devs want both to work and if they fail, they both fail in the same way. That's a slight nuance over the environments matching. I don't actually have the same VS version installed locally as what's in our PR/CI builds for example and generally that doesn't matter (I also can't run locally every matrix that the PR builds cover)

Reply:
I'm not sure which point in the text you are addressing. We're agreeing that devs want matching behavior. My main point is that the official build is truth so that if a bug only repros there, you definitely have a problem.